### DISCUSSION TITLE
Action based on metadata/annotations

### DISCUSSION THREAD

**Discussion by satishpasumarthi** [kubernetes-sigs/kubebuilder#3895] [Support] [answered]
I have an operator which watches for the status of pods, job and services. Whenever an annotation is noticed (applied through external factor kubectl patch...) on the job, the reconciler kicks in and performs the action I wanted (Delete a certain pod). But immediately after this action, I want to reset/update the annotation so that the reconciliation doesn't trigger infinitely. I tried to update the job's metadata/annotation from the reconciliation loop but it doesn't reflect when I do `kubectl get job` . Any suggestions?

**Comment by camilamacedo86** [comment, marked as answer]
Hi,

It seems that the problem you're facing is related to implementing non-idempotent logic. This approach does not align well with the operator pattern and controller-runtime principles. The foundational rule in these designs is to clearly declare the desired state and ensure that your implementations are idempotent. This means that no matter how many times the reconciliation loop executes, it should consistently produce the same results without unintended side effects.

So, you need think in something like:

Always, when has not the annotation to do A and etc. 
For further information, I would like to suggest you check out the following links. 
 
- https://book.kubebuilder.io/reference/good-practices#what-is-reconciliation-in-operators
- https://book.kubebuilder.io/getting-started

I hope that the info provided help you out.
I hope that you do not mind I am closing this one as sorted out. 
However, feel free to re-open if you need to.
